### PR TITLE
Feat/wt 2052 txn action required

### PR DESCRIPTION
**Pull request by ZacharyCouchman**
# Summary

[WT-2052](https://immutable.atlassian.net/browse/WT-2052)

Adding in the handling of transaction items in the Bridge so that withdrawals that are marked as pending can be identified and actioned.

# Customer Impact

This will allow users to see any withdrawals that are stuck in the Flow Rate queue and possibly action them if after the delay.

[WT-2052]: https://immutable.atlassian.net/browse/WT-2052?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

image

### CODE REVIEW

File: packages/checkout/widgets-lib/src/resources/text/textConfig.ts [hunk 1]
```diff
@@ -458,6 +458,11 @@ export const text = {
       layoutHeading: 'In progress',
       passportDashboard: 'View the full transaction history in your',
       status: {
+        withdrawalPending: {
```
Comment (dreamoftrees) [inline packages/checkout/widgets-lib/src/resources/text/textConfig.ts:461]: These need to be added into the widgets-lib locales en.JSON and other language files as well.

**Review by dreamoftrees — COMMENTED** [review]
Great work @ZacharyCouchman just some changes for translations needed. Happy to pair up with you if that is helpful.

File: packages/checkout/widgets-lib/src/components/Transactions/TransactionItem.tsx [hunk 2]
```diff
@@ -1,54 +1,46 @@
...
 type TransactionItemProps = {
   label: string
   details: {
     text: string,
     link: string,
     hash: string,
-  }
+  },
+  transaction: Transaction,
+  // token: TokenInfo,
```
Comment (sharonsheah) [inline packages/checkout/widgets-lib/src/components/Transactions/TransactionItem.tsx:22]: Is this still needed?

File: packages/checkout/widgets-lib/src/components/Transactions/TransactionItemWithdrawPending.tsx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+import { BridgeWidgetViews } from 'context/view-context/BridgeViewContextTypes';
+import { Transaction, TransactionStatus } from 'lib/clients/checkoutApiType';
+import { useMemo } from 'react';
+import { actionsContainerStyles, actionsLayoutStyles, containerStyles } from './transactionItemStyles';
+import { TransactionDetails } from './TransactionDetails';
+
+type TransactionItemWithdrawPendingProps = {
+  label: string
+  transaction: Transaction,
+  // token: TokenInfo,
```
Comment (sharonsheah) [inline packages/checkout/widgets-lib/src/components/Transactions/TransactionItemWithdrawPending.tsx:21]: Is this needed?

Comment (imx-mikhala) [inline packages/checkout/widgets-lib/src/components/Transactions/TransactionItemWithdrawPending.tsx:21]: is this prop going to be used?

File: packages/checkout/widgets-lib/src/components/Transactions/TransactionItem.tsx (hunk 2 above)
Comment (ZacharyCouchman) [reply packages/checkout/widgets-lib/src/components/Transactions/TransactionItem.tsx:22]: not required at the moment. I'll remove

File: packages/checkout/widgets-lib/src/resources/text/textConfig.ts (hunk 1 above)
Comment (ZacharyCouchman) [reply packages/checkout/widgets-lib/src/resources/text/textConfig.ts:461]: updated this now

**Review by sharonsheah — APPROVED** [review]

**Review by sharonsheah — APPROVED** [review]

**Review by imx-mikhala — DISMISSED** [review]

**Review by imx-mikhala — APPROVED** [review]

**Review by sharonsheah — APPROVED** [review]

**Pull request merged**